### ISSUE TITLE
[iOS] session(_:didReceiveMessage:) is interrupted by phone call 

### ISSUE THREAD

**Issue by chuanpham** [Rexios80/watch_connectivity#15]
I encountered the following problem, hoping to find a solution.

When a call comes in, session(_:didReceiveMessage:) is interrupted.

Has anyone here encountered this problem on iOS?

Thank you for your help.

**Comment by Rexios80** [comment]
Is that code running on the watch or the phone?

**Comment by chuanpham** [comment]
> Is that code running on the watch or the phone?

It's running on the phone

**Comment by Rexios80** [comment]
The code is in the plugin?

Can you make a sample project that reproduces the issue?

**Comment by chuanpham** [comment]
Thank you for your reply,

To reproduce the issue, you can simply create a timer on watchOs app, every 1 sec, you send a random message to iOS app. After that, whenever a call comes in, you'll see the `session(_:didReceiveMessage:)` is interrupted.

**Comment by Rexios80** [comment]
Isn't that what the example app does? Can you reproduce this with the example app?

**Comment by Rexios80** [comment]
What exactly is the error? Does it actually affect anything? Watch communication might just not work during a phone call.

**Comment by chuanpham** [comment]
The feature we are developing is based on the continuous sending and receiving of messages. Therefore, when receiving messages is interrupted due to an incoming call, it leads to some unexpected errors.

`Watch communication might just not work during a phone call.`: Do we have any solution to solve this issue? Thank you!

**Comment by Rexios80** [comment]
I've seen something like this before, but I'm pretty sure that was when answering the call on the watch itself. Does it happen when you answer the call on the phone as well?

**Comment by Rexios80** [comment]
And unfortunately I never did find a workaround for the messaging issue with watch phone calls

**Comment by chuanpham** [comment]
> I've seen something like this before, but I'm pretty sure that was when answering the call on the watch itself. Does it happen when you answer the call on the phone as well?

Yes it does happen when answering the call on the phone (It even happens when the phone rings as well).

**Comment by Rexios80** [comment]
Can you reproduce the issue with this repo?

https://github.com/vigilife/apple_watch_call_messaging_test

That's about as vanilla as it gets. If that app has the issue there's probably nothing we can do. 

Although I did have a thought: does your app have any background modes? The app I made this for has a bunch like Bluetooth, location, etc so maybe that's why we didn't observe the issue phone-side?

**Comment by chuanpham** [comment]
> Can you reproduce the issue with this repo?
> 
> https://github.com/vigilife/apple_watch_call_messaging_test
> 
> That's about as vanilla as it gets. If that app has the issue there's probably nothing we can do.
> 
> Although I did have a thought: does your app have any background modes? The app I made this for has a bunch like Bluetooth, location, etc so maybe that's why we didn't observe the issue phone-side?

The native iOS and WatchOS apps seemingly can communicate well with each other when a phone call is coming. May be it's the problem of Flutter itself (because Flutter currently does not support watchOS?).

**Comment by Rexios80** [comment]
Flutter not supporting watchOS has nothing to do with this. Do you get a specific exception that you can paste here?

**Comment by chuanpham** [comment]
> Flutter not supporting watchOS has nothing to do with this. Do you get a specific exception that you can paste here?

I see no specific exception. The `session(_:didReceiveMessage:)` is interrupted for a few minutes; after that, it works normally.

**Comment by Rexios80** [comment]
What do you mean by interrupted? Does Xcode catch an exception? Or is the method just not called during the phone call?

**Comment by chuanpham** [comment]
Xcode does not catch any exception, the method is just not called during the phone call.
